Generalized functions for getting, setting, and finding environment variables (#798)
NREL/GreenHEART: docs/getting_started/environment_variables.md

Reviewer: is it a yaml file for yml file here?

Author: its in the H2I root directory. I don't know much about setting environment variables with this method - I think that this method may require that the environment variables are set every time you re-activate your conda environment.

Reviewer: I know yaml and yml files are the same, but I think we should be consistent throughout this section

Author: good catch - I will clean it up!

Author: changed section header to "Set Environment Variables with .yml file"

(environment_variables:environment-variables)=
# Environment Variables

> H2Integrate can pull data from datasets accessible with API keys or user-specific tokens. Since API keys and tokens are unique to each user, these are accessed in H2Integrate as environment variables. These environment variables only need to be set to utilize their corresponding functionality. Some environment variables can also used to customize your workflow. The list of environment variables for different types of functionality are listed below:

Reviewer: This feels a little vague to me (how do you get user specific tokens?) but if it's something people generally should know, then I would leave it!

Author: I suppose the individual sections on API keys and such should explain that but idk - I guess I'm use "token" as a catch-all for any user-specific code/key/etc that may be needed to access publicly available datasets.

> H2Integrate can pull data from datasets accessible with API keys or user-specific tokens. Since API keys and tokens are unique to each user, these are accessed in H2Integrate as environment variables. These environment variables only need to be set to utilize their corresponding functionality. Some environment variables can also used to customize your workflow. The list of environment variables for different types of functionality are listed below:

Reviewer: ```suggestion
H2Integrate can pull data from datasets accessible with API keys or user-specific tokens. Since API keys and tokens are unique to each user, these are set individually and accessed by H2Integrate as environment variables. These environment variables need to be set to utilize their corresponding functionality. Some environment variables can also be used to customize your workflow. The list of environment variables that can be used by H2Integrate are listed below:
```

Author: I think that saying H2I sets the environment variables could be misleading in some weird way. With some of the methods to set environment variables, they're set with conda and accessed by H2I. With the .env file method, environment variables are loaded and set by H2I then accessed with H2I. I did make your other suggested changes.

> - [NLR Developer Network](environment_variables:nlr_developer)

Reviewer: I think it would be nice to add what kind of things you would need this for (i.e. wind and solar resource data)

Author: Updated first sentence to this: "H2Integrate can pull data (such as wind and solar resource data, feedstock prices, etc) from public datasets accessible with API keys or user-specific tokens"

    - `NLR_API_KEY`
    - `NLR_API_EMAIL`
- [EIA Natural Gas Cost Data](environment_variables:eia_ng)
    - `EIA_API_KEY`
- Customized Workflow
    - `RESOURCE_DIR`

```{note}

> Tips on debugging environment variable related errors or issues can be found [here](https://h2integrate.readthedocs.io/en/develop/misc_resources/debugging_environment_variables.html)

Reviewer 2: I built the docs locally and this link didn't work, why not use write it like: 

`[here](#env_var_debug:intro)`

and then add `(env_var_debug:intro)=` to the `debugging_environment_variables.md`? This would be more resilient to other branches that folks might be working on/looking at

Author: the link is broken because its not yet merged into develop, but I could give your way a try! I've had varying success with those types of links when they're in different files but I'll try it!

Author: I updated it with your suggestion @kbrunik

    NLR_API_EMAIL='email-for-api-key'
    ```
3. Save and close the ".env" file.

> (environment_variables:nlr_developer)=

Reviewer: This part seems to be in python code formatting (highlighting for and in) rather than markdown formatting. I don't known if that's just a GitHub thing here?

Reviewer 3: This is a myst markdown thing for creating cross-referencing tags that are able to be used easily from external files.

# NLR Developer Network Environment Variables

H2Integrate can pull weather resource datasets (e.g. data needed for wind or solar generation) automatically for a user-provided location.
To use resource datasets from the NLR developer network, you will need an NLR API key, which can be obtained from:
    [https://developer.nlr.gov/signup/](https://developer.nlr.gov/signup/).

> You will need to set the API key and the email you used to get the API key for downloading resource data from the NLR developer network. The 40 character API key is referred to in following sections as the value for the `NLR_API_KEY` environment variable. The email used to get the API key is referred to in the following sections as the value for the `NLR_API_EMAIL` environment variable.

Reviewer: ```suggestion
You will need to set the API key and the email you used to get the API key to download resource data from the NLR developer network. The 40 character API key is referred to in following sections as the value for the `NLR_API_KEY` environment variable. The email used to get the API key is referred to in the following sections as the value for the `NLR_API_EMAIL` environment variable.
```
